test/let_value_with_stop_token_test.cpp

```diff
@@ -72,7 +72,33 @@ struct NonInplaceStoppableIntReceiver : public UnstoppableSimpleIntReceiver {
   inplace_stop_source& source_;
 };
 
-TEST(LetWithStopToken, Simple) {
+struct LetWithStopToken : testing::Test {
+  struct DestructionCounter {
+    int value;
+    bool wasMoved{false};
+
+    DestructionCounter(DestructionCounter&& c) : value(c.value) {
+      c.wasMoved = true;
+    }
+
+    DestructionCounter(int value) : value(value) {}
+
+    static int& destroyCount() {
+      static int destroyCount = 0;
+      return destroyCount;
+    }
+
+    ~DestructionCounter() {
+      if (!wasMoved) {
+        destroyCount()++;
+      }
+    }
+  };
+
+  void SetUp() override { DestructionCounter::destroyCount() = 0; }
```

Comment:
For future, I'd do it this way.  I'm just gonna merge your diff, though.

```suggestion
  LetWithStopToken() noexcept { DestructionCounter::destroyCount() = 0; }
```